src/redux/actions/newArticle.js

```diff
@@ -48,13 +37,15 @@ export const newArticle = article => async dispatch => {
       tagsList
     });
     dispatch(articleCreated(response));
+    return history.push(`/articles/${response.data.article.slug}`);
   } catch (error) {
     const { message } = error.response.data;
     dispatch({ type: ARTICLE_ERROR, payload: message });
+    setTimeout(() => dispatch(clearResponse()), 3000);
```

Comment:
Bro, Did you tried to test Settimeout function?
if not, please have a look on [this blog](https://jestjs.io/docs/en/timer-mocks) post and see if you can do it.